internal/controllers/rest_controller.go

```diff
@@ -70,14 +87,18 @@ func (rc RESTController) CheckForAuthentication(w http.ResponseWriter, r *http.R
 	// check if the token is expired or signature invalid.
 	// if so, the OIDC flow will be started to recreate a token.
 	if authErrors.IsTokenExpired(err) || authErrors.IsTokenSignatureInvalid(err) {
+		if authType == "HeaderCredentialsHandler" {
+			rc.forbidden(w, err)
+			c.Abort()
+			return
+		}
```

Comment:
I didn't realize this block of code. Can you elaborate a bit the reason behind it?

Reply:
to identify what the request is...
if it's an authorization header, it shouldn't redirect to "login" again.